am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -68,27 +71,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
      * @param userId       (accessorId)
      * @param resourceId   resource id
      * @param resourceJson json
-     * @return resourceJson or null
+     * @return filtered resourceJson based on READ permissions or null if no READ permissions on resource
      */
     public FilterResourceResponse filterResource(String userId, String resourceId, JsonNode resourceJson) {
-        ExplicitAccessRecord explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
+        List<ExplicitAccessRecord> explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
             dao -> dao.getExplicitAccess(userId, resourceId));
 
-        if (explicitAccess == null) {
+        if (explicitAccess.isEmpty()) {
             return null;
         }
 
-        if (READ.isGranted(explicitAccess.getPermissions())) {
-            Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
-            attributePermissions.put("/", Permissions.fromSumOf(explicitAccess.getPermissions()));
+        Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
+        ObjectNode filteredResource = JsonNodeFactory.instance.objectNode();
 
+        explicitAccess.forEach(explicitAccessRecord -> {
+            attributePermissions.put(explicitAccessRecord.getAttribute(),
+                Permissions.fromSumOf(explicitAccessRecord.getPermissions()));
+            if (READ.isGranted(explicitAccessRecord.getPermissions())) {
+                filteredResource.set(explicitAccessRecord.getAttribute().replaceFirst("/", ""),
+                    resourceJson.at(JsonPointer.valueOf(explicitAccessRecord.getAttribute())));
+            }
+        });
+
+        if (filteredResource.toString().equals("{}")) {
+            return null;
```

Comment:
That is wrong. If filtered object is empty but permissions exist in DB the following should be returned:

```
{
  resourceId: "#1",
  data: null,
  permissions: {
    "/name": [DELETE]
  }
}
```

See: https://tools.hmcts.net/confluence/display/AM/005.1+Use+scenarios

Reply:
👍